### PR TITLE
Refactors OrderDetails component

### PR DESCRIPTION
**Pull request by jessicamarcus**
- Renamed ServiceFeeDisclosure.tsx and added research fee disclosure
- Moved and refactored OrderDetails as discussed in order to share across birth + death
- Added a CertificateType to types.ts

### CODE REVIEW

File: services-js/registry-certs/client/types.ts [hunk 1]
```diff
@@ -16,3 +16,5 @@ export interface DeathCertificateOrder
   extends NonNullable<
       LookupDeathCertificateOrder['deathCertificates']['order']
     > {}
+
+export type CertificateType = 'death' | 'birth';
```
Comment (fionawhim) [inline services-js/registry-certs/client/types.ts:20]: 👍

File: services-js/registry-certs/client/common/checkout/OrderDetails.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+  } else {
+    cart.setQuantity(TYPICAL_CERTIFICATE, 1);
+    cart.setQuantity(PENDING_CERTIFICATE, 3);
+    cart.setQuantity(NO_DATE_CERTIFICATE, 1);
+  }
+
+  return cart;
+}
+
+// const birthCertCart: Cart = {
```
Comment (fionawhim) [inline services-js/registry-certs/client/common/checkout/OrderDetails.stories.tsx]: ok to just delete this?

File: services-js/registry-certs/client/common/checkout/OrderDetails.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+const loadingCart = makeCart(true);
+const deathCart = makeCart(false);
+
+storiesOf('Common Components/Order Details', module)
+  .add('OrderDetailsDropdown: loading', () => (
+    <OrderDetailsDropdown
+      orderType="death"
+      certificateQuantity={loadingCart.size}
+    >
+      <DeathOrderDetails cart={loadingCart} />
```
Comment (fionawhim) [inline services-js/registry-certs/client/common/checkout/OrderDetails.stories.tsx:52]: Good refactor.

File: services-js/registry-certs/client/common/checkout/OrderDetails.tsx [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+              {certificateCostString} = ${(
+                calculateCreditCardCost(certificateCost, quantity).subtotal /
+                100
+              ).toFixed(2)}{' '}
+              + service fee*
+              {this.props.hasResearchFee && (
+                <>
+                  <span> + research fee</span>
+                  {/* dagger character looks strange when italicized */}
+                  <span style={{ fontStyle: 'normal' }}>†</span>
```
Comment (fionawhim) [inline services-js/registry-certs/client/common/checkout/OrderDetails.tsx:164]: bztf

**Review by fionawhim — APPROVED** [review]

File: services-js/registry-certs/client/common/checkout/OrderDetails.stories.tsx (hunk 2 above)
Comment (jessicamarcus) [reply services-js/registry-certs/client/common/checkout/OrderDetails.stories.tsx]: Thanks for the catch!

File: services-js/registry-certs/client/common/checkout/OrderDetails.tsx (hunk 4 above)
Comment (jessicamarcus) [reply services-js/registry-certs/client/common/checkout/OrderDetails.tsx:164]: 🔪 🔪 🔪

**Pull request merged**